secrets/Makefile

```diff
@@ -0,0 +1,88 @@
+    build/concent-secrets/secrets.py                          \
+    build/concent-secrets/var-secret.yml                      \
+    build/concent-secrets/nginx-proxy-ssl.crt                 \
+    build/concent-secrets/nginx-proxy-ssl.key                 \
+    build/concent-secrets/nginx-storage-ssl.crt               \
+    build/concent-secrets/nginx-storage-ssl.key               \
+    build/templates/nginx-proxy-secrets.yml                   \
+    build/templates/nginx-storage-secrets.yml                 \
+    build/templates/control-db-secrets.yml                    \
+    build/templates/storage-db-secrets.yml                    \
+    build/templates/django-admin-fixture.yml                  \
+    build/templates/concent-api-secrets.yml                   \
+    build/apply-secrets.sh                                    \
+    build/delete-secrets.sh
+
+
+ifndef CLUSTER
+    $(error CLUSTER parameter not specified)
+endif
+
+$(info CLUSTER=$(CLUSTER))
+ifndef CONCENT_SECRET_DIR
+    CONCENT_SECRET_DIR := $(abspath ../../concent-secrets/$(CLUSTER))
+endif
+$(info CONCENT_SECRET_DIR=$(CONCENT_SECRET_DIR))
+
+all: cluster-secrets
+
+cluster-secrets: build/ $(CLUSTER_SECRETS)
+
+build/:
+	mkdir --parents "$@"
+
+# Static file
+build/%: %
+	cp --archive --parents "$<" build/
+
+# Templates
+build/%: %.j2 build/concent-secrets/var-secret.yml ../tools/render-jinja-template.sh
+	../tools/render-jinja-template.sh build/concent-secrets/var-secret.yml "$<" "$@"
+
+# Secret file
+build/concent-secrets/%: $(CONCENT_SECRET_DIR)/%
+	mkdir --parents $(dir $@)
+	cp --archive "$<" $(dir $@)
+
+build/templates/nginx-%-secrets.yml: build/concent-secrets/nginx-%-ssl.crt build/concent-secrets/nginx-%-ssl.key
+	mkdir --parents $(dir $@)
+	kubectl create secret generic $(notdir $(basename $@))      \
+		--from-file=$(notdir $<)=$<                             \
+		--from-file=$(notdir $(word 2,$^))=$(word 2,$^)         \
+		--dry-run                                               \
+		-o yaml > $@
+
+# Need update to `nginx-storage.crt` after separate to cluster control and storage
+build/templates/concent-api-secrets.yml: build/concent-secrets/secrets.py build/concent-secrets/nginx-proxy-ssl.crt
+	kubectl create secret generic $(notdir $(basename $@))      \
+		--from-file=$(notdir $<)=$<                             \
+		--from-file=$(notdir $(word 2,$^))=$(word 2,$^)         \
+		--dry-run                                               \
+		-o yaml > $@
+
+build/templates/django-admin-fixture.yml: build/secrets/django-admin-fixture.yaml
+	kubectl create secret generic $(notdir $(basename $@))      \
+		--from-file=$(notdir $<)=$<                             \
+		--dry-run                                               \
+		-o yaml > $@
+
+
+build/templates/%-db-secrets.yml: build/vars-with-%-database-config-and-secrets.yml templates/db-secrets.yml.j2 ../tools/render-jinja-template.sh
+	../tools/render-jinja-template.sh "$<" "templates/db-secrets.yml.j2" "$@"
```

Comment:
I thought we'd be able to remove `vars-with-%-database-config-and-secrets.yml` and `%-database-configuration.yml`. I see the only reason they're still here is that you need to render `db-secrets.yml.j2` twice, with different passwords.

Let's simplify this. Create two separate templates: `storage-db-secrets.yml.j2` and `control-db-secrets.yml.j2`. This way you'll be able to render them using only `var-secret.yml` without the need for the rules below.

Reply:
A little bit. But at the same time it removes more code than we're repeating so it's worth it in my opinion.